core/scheduler/scheduler.go

```diff
@@ -317,7 +325,12 @@ func (s *Scheduler) resolveProDuties(ctx context.Context, slot slot, vals valida
 			continue
 		}
 
+		// TODO: replace if false with check for builder api flag
```

Comment:
We have 2 options for checking for the builder api flag

in the resolveDuties function and implement the resolveBuilderProDuties function in this PR or have one generic resolveProDuties and have the check for the builder api flag in there

In the future i think you'll probably want to schedule both and only execute 1

Some notes from flashbots

- MEV Boost has a BUILDER_PROPOSAL_DELAY_TOLERANCE which is the deadline within the slot for a builder to have proposed a block (External Block Build)
- After this deadline the “External Block Build” will be aborted in favour of a “Local Block Build”
- From MEV Boost documentation, both “External Block Build” and “Local Block Build” should begin as soon as they are able to. The Charon client must ensure one or the other NOT both are submitted.